web/src/store/publicDandisets.js

```diff
@@ -0,0 +1,44 @@
+import girderRest from '@/rest';
+
+const DANDISETS_PER_PAGE = 8;
+
+export default {
+  namespaced: true,
+  state: {
+    // search settings
+    sort: {
+      field: 'created',
+      direction: 1,
+    },
+    page: 1,
+
+    // retrieved from server
+    dandisets: [],
+    pages: 0,
+  },
+  mutations: {
+    setSearchSettings(state, { sort }) {
+      state.sort = sort;
+      state.page = 1;
+      this.dispatch('publicDandisets/reload');
```

Comment:
Mutations aren't supposed to dispatch actions, only actions should, and mutations are supposed to very narrowly change properties of state. So the action of say `changeSort` would commit the mutation of setting `state.sort` and `state.page`, and then dispatch this action.

In general, I don't think you're supposed to reference `this` inside of a vuex store, I don't think what it exactly references is defined, which is why ideally in the `vuex` world you wouldn't be able to dispatch actions here, as `dispatch` isn't exposed through passed arguments.